src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -987,13 +987,13 @@ COptTasks::PrintMissingStatsWarning(CMemoryPool *mp, CMDAccessor *md_accessor,
 	if (0 < rel_stats->Size())
 	{
 		int length = NAMEDATALEN * rel_stats->Size() + 200;
-		char msgbuf[length];
+		std::vector<char> msgbuf(length);
```

Comment:
What about using some string class instead of vector? For example std::string

Reply:
I'm not sure that it's safe to pass std::string::data() like I did here

Reply:
What about something like
```c
if (0 == ul)
{
    oss << "One or more columns in the following table(s) do not have statistics: ";
}
else
{
    oss << ", ";
}
```
?